pkg/agent/conversation.go

```diff
@@ -230,9 +230,64 @@ func (a *Conversation) RunOneRound(ctx context.Context, query string) error {
 				return fmt.Errorf("building tool call: %w", err)
 			}
 
+			// Check if this is a kubectl or bash command that might be interactive
+			isInteractive := false
+			var errMsg string
+			if call.Name == "kubectl" || call.Name == "bash" {
```

Comment:
I wonder if we can introduce a method in the `Tool` interface called `IsInteractive(args) bool` to delegate this logic to individual tools to keep the core agentic loop tool agnostic.